test/evm/foundry/fork/BlacklistedRelayerRecipient.t.sol

```diff
@@ -0,0 +1,164 @@
+// SPDX-License-Identifier: Unlicense
+pragma solidity ^0.8.0;
+
+import { Test } from "forge-std/Test.sol";
+import { MockSpokePool } from "../../../../contracts/test/MockSpokePool.sol";
+
+import "@openzeppelin/contracts-upgradeable/token/ERC20/IERC20Upgradeable.sol";
+
+// Define a minimal interface for USDT. Note USDT does NOT return anything after a transfer.
+interface IUSDT {
+    function balanceOf(address account) external view returns (uint256);
+
+    function transfer(address recipient, uint256 amount) external;
+
+    function transferFrom(
+        address from,
+        address to,
+        uint256 value
+    ) external;
+
+    function addBlackList(address _evilUser) external;
+
+    function getBlackListStatus(address _evilUser) external view returns (bool);
+}
+
+// Define a minimal interface for USDC. Note USDC returns a boolean after a transfer.
+interface IUSDC {
+    function balanceOf(address account) external view returns (uint256);
+
+    function transfer(address recipient, uint256 amount) external returns (bool);
+
+    function transferFrom(
+        address from,
+        address to,
+        uint256 value
+    ) external returns (bool);
+
+    function blacklist(address _account) external;
+
+    function isBlacklisted(address _account) external view returns (bool);
+}
+
+contract MockSpokePoolTest is Test {
+    MockSpokePool spokePool;
+    IUSDT usdt;
+    IUSDC usdc;
+
+    address largeUSDTAccount = 0x99C9fc46f92E8a1c0deC1b1747d010903E884bE1;
+    address largeUSDCAccount = 0x37305B1cD40574E4C5Ce33f8e8306Be057fD7341;
+    uint256 seedAmount = 10_000 * 10**6;
+
+    address recipient1 = address(0x6969691111111420);
+    address recipient2 = address(0x6969692222222420);
+
+    function setUp() public {
+        spokePool = new MockSpokePool(address(0x123));
+        // Create an instance of USDT & USDCusing its mainnet address
+        usdt = IUSDT(address(0xdAC17F958D2ee523a2206206994597C13D831ec7));
+        usdc = IUSDC(address(0xA0b86991c6218b36c1d19D4a2e9Eb0cE3606eB48));
+
+        // Impersonate a large USDT & USDC holders and send tokens to spokePool contract.
+        assertTrue(usdt.balanceOf(largeUSDTAccount) > seedAmount, "Large USDT holder has less USDT than expected");
+        assertTrue(usdc.balanceOf(largeUSDCAccount) > seedAmount, "Large USDC holder has less USDC than expected");
+
+        vm.prank(largeUSDTAccount);
+        usdt.transfer(address(spokePool), seedAmount);
+        assertEq(usdt.balanceOf(address(spokePool)), seedAmount, "Seed transfer failed");
+
+        vm.prank(largeUSDCAccount);
+        usdc.transfer(address(spokePool), seedAmount);
+        assertEq(usdc.balanceOf(address(spokePool)), seedAmount, "USDC seed transfer failed");
+    }
+
+    function testStandardRefundsWorks() public {
```

Comment:
This is a really cool usage of foundry fork tests to verify that we're compatible with USDT in production! This doesn't run in CI, right?

Reply:
nope, you need to run it sepertly.